pkg/runtime/scheduler/scheduler.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright 2024 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheduler
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	schedulerv1pb "github.com/dapr/dapr/pkg/proto/scheduler/v1"
+	"github.com/dapr/dapr/pkg/scheduler"
+	schedulerclient "github.com/dapr/dapr/pkg/scheduler/client"
+	"github.com/dapr/dapr/pkg/security"
+	"github.com/dapr/kit/logger"
+	"google.golang.org/grpc"
+)
+
+var log = logger.NewLogger("dapr.runtime.scheduler")
+
+type Scheduler struct {
+	Addresses    []string
+	SidecarAddr  string
+	SidecarPort  int
+	SidecarNS    string
+	SidecarAppID string
```

Comment:
```suggestion
	Namespace    string
	AppID string
```

Reply:
I realize now I want to rm this struct. Instead of using this, from runtime I'll pass in the sidecarConnDetails struct and scheduler addresses and sec separately.